_episodes/08-putting-it-all-together.md

```diff
@@ -86,209 +89,226 @@ also change the name of the columns in the DataFrame like this:
+> doing some of the labor of visualizing plots for you. But, not every one of your collaborators
+> will be using a Jupyter Notebook. The .show() command allows you to visualize plots
+> when working at the command line, with a script, or at the iPython interpreter. In the 
+> previous example, adding  `plt.show()` after the creation of the plot will enable your 
+> colleagues who aren't using a Jupyter notebook to reproduce your work on their platform.
+{: .callout}
 
-    plt.show()
+~~~
+fig, ax = plt.subplots()  # initiate an empty figure and axis matplotlib object
+ax.plot(x, y, '-')
 ~~~
 {: .language-python}
 
-The function `legend()` adds a legend to the figure, and the optional keyword
-arguments change its style. By default [typing just `plt.legend()`], the legend
-is on the upper right corner and has no shadow.
+![line plot](../fig/08_line_plot.png)
 
- The functions `xlabel`, `ylabel`, `title`, `legend`, and many others create text labels. It is good to know that, in addition to the plain text, you may use mathematical notation using a subset of LaTeX language. See [this link](https://matplotlib.org/users/mathtext.html) for more information.
+Although the latter requires a little bit more code to create the same plot, the advantage is that we now have **full control** of where the plot axis are placed, and we can easily add new items or, for example, more than one axis to the figure and adapt the labels:
 
+~~~
+fig, ax1 = plt.subplots() # prepare a matplotlib figure
+ax1.plot(x, y, '-')
 
-Like MATLAB, pyplot is stateful; it keeps track of the current figure and
-plotting area, and any plotting functions are directed to those axes. To make
-more than one figure, we use the command `plt.figure()` with an increasing
-figure number inside the parentheses:
+# adapt the labels
+ax1.set_ylabel('y')
+ax1.set_xlabel('x')
 
+# add an additional ax to the figure
+ax2 = fig.add_axes([0.2, 0.5, 0.4, 0.3]) # inset axes
+ax2.plot(x, y*2, 'r-')
 ~~~
-    # This is the first figure
-    plt.figure(1)
-    plt.plot(t, t, 'r--', label='linear')
+{: .language-python}
+
+![Additional axis to line plot](../fig/08_line_plot_inset.png)
 
-    plt.legend(loc='upper left', shadow=True, fontsize='x-large')
-    plt.title('This is figure 1')
+### Link matplotlib, Pandas and plotnine
 
-    plt.show()
+The **Pandas and plotnine packages create matplotlib objects** as well. Hence, using the object based approach provides a consistent workflow and interaction between these packages:
 
-    # This is a second figure
-    plt.figure(2)
-    plt.plot(t, t**2, 'bs-', label='square')
+~~~
+fig, ax1 = plt.subplots() # prepare a matplotlib figure
 
-    plt.legend(loc='upper left', shadow=True, fontsize='x-large')
-    plt.title('This is figure 2')
+surveys.plot("hindfoot_length", "weight", kind="scatter", ax=ax1)
 
-    plt.show()
+# Provide further adaptations with matplotlib:
```

Comment:
This is not a good example for a comment, IMHO. Comments should be specific to the code, so here it should be something like `Set a label for the x axis`